src/main/java/net/smartcosmos/cluster/auth/SmartCosmosUserCache.java

```diff
@@ -0,0 +1,58 @@
+package net.smartcosmos.cluster.auth;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.cache.SpringCacheBasedUserCache;
+
+import net.smartcosmos.security.SecurityResourceProperties;
+import net.smartcosmos.security.user.SmartCosmosCachedUser;
+
+/**
+ *
+ */
+@Slf4j
+public class SmartCosmosUserCache extends SpringCacheBasedUserCache {
+
+    private Duration cachedUserExpirationTime = Duration.millis(SecurityResourceProperties.DEFAULT_CACHED_USER_KEEP_ALIVE_SECS * 1000);
+
+    /**
+     * Create an instance of the user cache.
+     * <p>
+     * This cache is a simple cache implementation that adds expiration to the {@code SpringCacheBasedUserCache}.  On fetch if
+     * the time a a user has been in the cache exceeds the expiration time the user is ejected from the cache and a null is returned.
+     * </p>
+     *
+     * @param cache the embedded cache inside this user cache
+     * @param securityResourceProperties configuration properties containing the number of millseconds a user is kept in the cache
+     * @throws Exception something has gone horribly wrong
+     */
+    @Autowired
+    public SmartCosmosUserCache(Cache cache, SecurityResourceProperties securityResourceProperties) throws Exception {
+
+        super(cache);
+        this.cachedUserExpirationTime = Duration.millis(securityResourceProperties.getCachedUserKeepAliveSecs() * 1000);
+        log.debug("User cache initialized.");
+    }
+
+    @Override
+    public UserDetails getUserFromCache(String username) {
+
+        SmartCosmosCachedUser cachedUser = (SmartCosmosCachedUser) super.getUserFromCache(username);
```

Comment:
I am re-reading comments, I missed @jbauerrfid comment earlier. We spoke of this in the past and it is my opinion we should not allow the same id in different tenants.  This is one reason to use email addresses.  We would do better to implement support for one user across multiple tenant spaces than implementing logic to allow the same id in different tenants.